src/main/java/lotto/service/Calculator.java

```diff
@@ -0,0 +1,62 @@
+package lotto.service;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lotto.domain.BonusNumber;
+import lotto.domain.Lotto;
+import lotto.domain.PurchaseAmount;
+import lotto.domain.WinningInformation;
+import lotto.domain.WinningNumbers;
+import lotto.domain.WinningPrize;
+
+public class Calculator {
+
+    public int matchCount(Lotto issuedLotto) {
+        List<Integer> issuedNumbers = issuedLotto.getNumbers();
+        Set<Integer> winningNumbers = new HashSet<>(WinningNumbers.getInstance().getWinningNumbers());
+
+        int matchCount = 0;
+
+        for (Integer issuedNumber : issuedNumbers) {
```

Comment:
```suggestion
        for (int issuedNumber : issuedNumbers) {
```

wrapping 클래스는 최대한 primitive로 변경하여 연산하는 것이 unboxing 과정을 줄여 좋습니다.